### PR TITLE
Add relaxed standard shape assertion

### PR DESCRIPTION
**Pull request by jungpark-mlir**
Reiterate the assertion on the standard shape but relax it for the multibroadcast ops deliberately inserted to explicit the broadcast.

### CODE REVIEW

File: src/targets/gpu/mlir.cpp [hunk 1]
```diff
@@ -194,19 +194,20 @@ struct mlir_program
         return result;
     }
 
-    MlirType make_tensor(const shape& s) const
+    MlirType make_tensor(const shape& s, bool bStandard) const
```
Comment (pfultz2) [inline src/targets/gpu/mlir.cpp:197]: Follow C++ naming for variable names.

File: src/targets/gpu/mlir.cpp [hunk 2]
```diff
@@ -498,7 +499,7 @@ struct mlir_program
             auto ops  = create_operation_state(name);
             ops.add_attribute_value(get_operator_value(ins->get_operator()));
             if(ins->name() != "@return")
-                ops.add_results({get_shape(ins)});
+                ops.add_results({get_shape(ins)}, ins->name() != "multibroadcast");
```
Comment (pfultz2) [inline src/targets/gpu/mlir.cpp:502]: We would need to this for more then just broadcast. Transpose will have a non-standard shape as well and output of many operators will propagate the permutation as well so they all will be non-standard.

File: src/targets/gpu/mlir.cpp [hunk 3]
```diff
@@ -369,9 +370,9 @@ struct mlir_program
             return *this;
         }
 
-        mlir_operation_state& add_results(const std::vector<shape>& outputs)
+        mlir_operation_state& add_results(const std::vector<shape>& outputs, bool bStandard)
         {
-            auto x = prog->make_tensors(outputs);
+            auto x = prog->make_tensors(outputs, bStandard);
```
Comment (pfultz2) [inline src/targets/gpu/mlir.cpp:375]: I would rather convert the shape to standard instead of trying to pass a boolean through all of this.

Comment (jungpark-mlir) [reply src/targets/gpu/mlir.cpp:375]: Do you mean, having something like
shape::get_standardize()
which recalculates stride for the lens, ignoring the inferred shape?

That could workaround this. little hacky.

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**